tests/js/dataMock/analytics.js

```diff
@@ -0,0 +1,311 @@
+}
+
+/**
+ * Converts a string to a date object and returns it.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string} dateString Date string.
+ * @return {Date} A date object.
+ */
+function stringToDate( dateString ) {
+	return new Date( `${ dateString } 00:00:00` );
+}
+
+/**
+ * Gets metric key.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string|Object} metric Metric name or object.
+ * @return {string} Metric key.
+ */
+function getMetricKey( metric ) {
+	return metric?.expression || metric.toString();
+}
+
+/**
+ * Gets metric type.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string|Object} metric Metric name or object.
+ * @return {string} Type of the metric.
+ */
+function getMetricType( metric ) {
+	return ANALYTICS_METRIC_TYPES[ getMetricKey( metric ) ];
+}
+
+/**
+ * Generates and returns metric values.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string} type Metric type.
+ * @param {number} max  Maximum number of values to generate.
+ * @return {Array.<Object>} Array of metric values.
+ */
+function generateMetricValues( type, max ) {
+	const metrics = [];
+
+	for ( let i = 0; i < max; i++ ) {
+		const values = [];
+
+		switch ( type ) {
+			case 'INTEGER':
+				values.push( faker.random.number( { min: 0, max: 100 } ).toString() );
+				break;
+			case 'PERCENT':
+				values.push( faker.random.float( { min: 0, max: 100 } ).toString() );
+				break;
+			case 'TIME':
+				values.push( faker.random.number( { min: 0, max: 3600 } ).toString() ); // 1 hour max.
+				break;
+			case 'CURRENCY':
+				values.push( faker.random.float( { min: 0, max: 10000 } ).toString() ); // $10k max.
+				break;
+		}
+
+		metrics.push( { values } );
+	}
+
+	return metrics;
+}
+
+/**
+ * Sorts report rows and returns it.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Array.<Object>}        rows    Array of rows to sort.
+ * @param {Array.<Object>}        metrics Array of report metrics.
+ * @param {Object|Array.<Object>} orderby Sorting options.
+ * @return {Array.<Object>} Sorted rows.
+ */
+function sortRows( rows, metrics, orderby ) {
+	let sorted = rows;
+
+	const orders = Array.isArray( orderby ) ? orderby : [ orderby ];
+	for ( const order of orders ) {
+		const direction = order?.sortOrder === 'DESCENDING' ? -1 : 1;
+		const index = metrics.findIndex( ( metric ) => getMetricKey( metric ) === order?.fieldName );
+		if ( index < 0 ) {
+			continue;
+		}
+
+		sorted = sorted.sort( ( a, b ) => {
+			let valA = parseFloat( a.metrics[ index ]?.values?.[ 0 ] );
+			if ( Number.isNaN( valA ) ) {
+				valA = 0;
+			}
+
+			let valB = parseFloat( b.metrics[ index ]?.values?.[ 0 ] );
+			if ( Number.isNaN( valB ) ) {
+				valB = 0;
+			}
+
+			return ( valA - valB ) * direction;
+		} );
+	}
+
+	return sorted;
+}
+
+/**
+ * Generates mock data for Analytics reports.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object} args Report options.
+ * @return {Array.<Object>} An array with generated report.
+ */
+export function getAnalyticsMockResponse( args ) {
+	faker.seed( getObjectHash( args ) );
```

Comment:
It helps us to control the randomness on a per widget basis because the seed number is generated from the provided arguments. I have added a few lines of comments to make it more clear.

Reply:
Ah, I get it now. Thanks for the explanation here and the comment in the code 👍🏻